src/main/java/com/direwolf20/buildinggadgets/common/construction/modes/CircleMode.java

```diff
@@ -0,0 +1,71 @@
+package com.direwolf20.buildinggadgets.common.construction.modes;
+
+import com.direwolf20.buildinggadgets.common.construction.ModeUseContext;
+import com.direwolf20.buildinggadgets.common.construction.XYZ;
+import com.direwolf20.buildinggadgets.common.schema.BoundingBox;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CircleMode extends Mode {
+    private final boolean filled;
+    public CircleMode(boolean isExchanging, boolean filled) {
+        super(filled ? "circle" : "hollow_circle", isExchanging);
+        this.filled = filled;
+    }
+
+    @Override
+    List<BlockPos> collect(ModeUseContext context, PlayerEntity player, BlockPos start) {
+        List<BlockPos> coordinates = new ArrayList<>();
+        int radius = context.getRange();
+        Direction direction = context.getHitSide();
+
+        int x = radius;
+        int y = 0;
+        int correction = 1 - x;
+
+        while(x >= y) {
+            if (this.filled) {
+                for (int i = -x; i <= x; i++) {
+                    coordinates.add(this.getBlockPosition(start, i, y, direction));
+                    if (y != 0) coordinates.add(this.getBlockPosition(start, i, -y, direction));
```

Comment:
New line for this if you could, I prefer it to be wrapped with `{` `}` but I'm not to bothered.